### DISCUSSION TITLE
network/iptables

### DISCUSSION THREAD

**Discussion by giscus** [isno/theByteBook#217] [General]
# network/iptables

构建大规模高可用的分布式系统

https://www.thebyte.com.cn/network/iptables.html

**Comment by binlongzhang** [comment]
数据包具体经过的表、链的关系和顺序如图 3-6 所示？是3-2？

**Reply by isno** [reply]
谢谢，我写错了，已修改。

**Comment by xiongyunhua-star** [comment]
kube-proxy使用ipvs时， 不支持DNAT转换， 仍然依赖iptables的dnat吗？

**Reply by xiongyunhua-star** [reply]
sh-4.4# ipvsadm   --list --numeric
IP Virtual Server version 1.2.1 (size=4096)
Prot LocalAddress:Port Scheduler Flags
  -> RemoteAddress:Port           Forward Weight ActiveConn InActConn
TCP  172.21.64.2:30002 rr
  -> 172.21.33.16:8080            Masq    1      0          0
TCP  172.21.64.2:30003 rr
  -> 172.21.33.16:8443            Masq    1      0          0
TCP  172.21.64.2:30009 rr
  -> 172.21.33.16:8443            Masq    1      0          0
TCP  172.21.64.2:30031 rr
  -> 172.21.32.220:5913           Masq    1      0          0
TCP  172.21.64.2:30035 rr
  -> 172.21.32.89:80              Masq    1      0          33       这些不都是DNAT规则吗？

**Reply by isno** [reply]
请求过来时，IPVS 负责 DNAT 转换，不依赖 iptables的 dnat。

**Reply by xiongyunhua-star** [reply]
嗯嗯，我理解错作者原文的意思了，感谢解答

**Comment by winlans** [comment]
iptables 共有 5 规则表 => iptables 共有 5个 规则表

**Comment by zjyyzhengli** [comment]
REJECT：功能与 DROP 类似，只不过它会通过 ICMP 协议给发送端返回错误信息，比如 Destination network unreachable 或者 Destination host。
-- 最后的“Destination host”似乎缺少了一些内容，是否是“Destination host **unreachable**”？
